pkg/rpcserver/middleware.go

```diff
@@ -0,0 +1,53 @@
+package rpcserver
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const requiredHeader = "sec-x-stripe-cli"
+
+// Only allow requests from clients with a sec-x-stripe-cli header
+func authorize(ctx context.Context) error {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return status.Errorf(codes.InvalidArgument, "Retrieving metadata failed")
+	}
+
+	if _, ok := md[requiredHeader]; !ok {
```

Comment:
how does that header work? will we ever care about the header value and need to validate, or do we just want the header's presence? In which case, why is this necessary?

Reply:
It's necessary to prevent malicious websites from making requests to the user's daemon. Their browser won't allow adding a header with prefix `sec-`, but other clients like vscode-stripe can. I can dm you with the details if you'd like.

Reply:
Playing devil's advocate: Are we concerned about other malicious clients?

Please DM details, I'd love to read more on that!